frontends/election-manager/src/lib/backends/storage_backend.ts

```diff
@@ -157,30 +159,39 @@ export class ElectionManagerStoreStorageBackend
     }
   }
 
-  async setCastVoteRecordFiles(
-    newCastVoteRecordFiles: CastVoteRecordFiles
-  ): Promise<void> {
-    if (newCastVoteRecordFiles === CastVoteRecordFiles.empty) {
-      await this.removeStorageKeyAndLog(cvrsStorageKey, 'Cast vote records');
-      await this.removeStorageKeyAndLog(
-        isOfficialResultsKey,
-        'isOfficialResults flag'
-      );
-    } else {
-      await this.setStorageKeyAndLog(
-        cvrsStorageKey,
-        newCastVoteRecordFiles.export(),
-        'Cast vote records'
-      );
+  async addCastVoteRecordFile(newCastVoteRecordFile: File): Promise<void> {
+    const loadElectionResult =
+      await this.loadElectionDefinitionAndConfiguredAt();
+
+    if (!loadElectionResult) {
+      throw new Error('Cannot add CVR files without an election definition.');
     }
+
+    const loadCastVoteRecordFilesResult =
+      (await this.loadCastVoteRecordFiles()) ?? CastVoteRecordFiles.empty;
+
+    const newCastVoteRecordFiles = await loadCastVoteRecordFilesResult.add(
+      newCastVoteRecordFile,
+      loadElectionResult.electionDefinition.election
+    );
+
+    await this.setStorageKeyAndLog(
```

Comment:
so storage backend is still just storing things in local storage? Do we need to add an entirely new backend type for saving in the database or will this just be updated to talk to admin-service at some point?

Reply:
The `ElectionManagerStoreAdminBackend` extends this one and has bits overridden to handle talking to the admin service.